config/deploy.rb

```diff
@@ -32,9 +32,27 @@
 
 set :keep_releases, 5
 
+set :npm_target_path, -> { release_path.join('frontend') } # default not set
+set :npm_flags, ''
+
 namespace :deploy do
   after :finishing, 'deploy:cleanup'
   after 'deploy:publishing', 'deploy:restart'
 end
 
-set :rvm_ruby_version, '2.4.1'
\ No newline at end of file
+namespace :npm do
+  after 'npm:install', 'npm:build'
+
+  task :build do
+    on roles fetch(:npm_roles) do
+      within fetch(:npm_target_path, release_path) do
+        with fetch(:npm_env_variables, {}) do
+          upload! "frontend/.env.#{fetch(:stage)}", "#{current_path}/frontend/.env"
```

Comment:
does this upload a local `.env` file? could we not do it following the same principle as the backend .env file, which is kept on it's target environment between releases?

Reply:
:+1: